File: python-iot-gateway/consumer.py
import json
from kafka import KafkaConsumer

if __name__ == "__main__":
    # kafka consumer
    consumer = KafkaConsumer(
        "messages",
        bootstrap_servers = 'localhost:9092',
        auto_offset_reset = 'earliest'
    )

    for message in consumer: 
        print(json.loads(message.value))

File: node-backend/kafka/consumer.js
import { Kafka } from "kafkajs";
import dataStorage from "./dataStorage.js";

// kafka client
const kafka = new Kafka({
  clientId: "simple-producer-consumer-application",
  //brokers: ["broker:29092"],
  brokers: ["localhost:9092"],
});

const startConsumer = async () => {
  const consumer = kafka.consumer({ groupId: "simple-group" });

  await consumer.connect();
  await consumer.subscribe({ topic: "messages", fromBeginning: true });

  await consumer.run({
    eachMessage: async ({ topic, partition, message }) => {
      
      const responseMessage = await dataStorage(JSON.parse(message.value))
      console.log({
        responseMessage,
        topic,
        partition,
      });
    },
  });
};


export default startConsumer;